backend/tests/test_modifyrepo.py

```diff
@@ -547,7 +547,11 @@ def test_comps_present(self, f_third_build):
             pass
 
         assert call_copr_repo(chrootdir)
-        name = glob.glob(os.path.join(chrootdir, "repodata", "*-comps.xml.gz"))
+
+        f39 = distro.id() == "fedora" and int(distro.version()) >= 39
+        compression = "zst" if f39 else "gz"
+        pattern = "*-comps.xml.{0}".format(compression)
+        name = glob.glob(os.path.join(chrootdir, "repodata", pattern))
```

Comment:
Just a test, but what about to check `*-comps.xml.*` pattern, and then `name.endswith(".gz")`?

Reply:
Ah, you have it in code below :-) and it is better, it won't complain once rhel moves to zstd, etc.